File: LeetCode/80. Remove Duplicates from Sorted Array II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        d = dict()
        cc =0
        for i in range(0, len(nums)):
            if nums[i] not in d.keys():
                d[nums[i]] = 1
            else:
                d[nums[i]] = 2

        for k,v in d.items():
            for j in range(v):
                nums[cc] = k
                cc+=1
        del nums[cc:]

File: LeetCode/80. Remove Duplicates from Sorted Array II.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length < 3) {
            return nums.length;
        }
        int prev = 2;
        for (int i = 2; i < nums.length; i++) {
            if (nums[i] != nums[prev - 2]) {
                nums[prev] = nums[i];
                prev++;
            }
        }
        return prev;
    }
}
